ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -369,14 +370,26 @@ getBlockIdsImpl VolatileDBEnv{..} = do
 
 getSuccessorsImpl :: forall m blockId. (MonadSTM m, Ord blockId)
                   => VolatileDBEnv m blockId
-                  -> m (Maybe blockId -> Set blockId)
+                  -> STM m (Maybe blockId -> Set blockId)
 getSuccessorsImpl VolatileDBEnv{..} = do
-    mSt <- atomically (readTMVar _dbInternalState)
+    mSt <- readTMVar _dbInternalState
     case mSt of
-        Nothing -> EH.throwError _dbErr $ UserError ClosedDBError
+        Nothing -> EH.throwError' _dbErrSTM $ UserError ClosedDBError
         Just st -> return $ \blockId ->
             fromMaybe Set.empty (Map.lookup blockId (_currentSuccMap st))
 
+getPredecessorImpl :: forall m blockId. (MonadSTM m, Ord blockId, HasCallStack)
+                   => VolatileDBEnv m blockId
+                   -> STM m (blockId -> Maybe blockId)
+getPredecessorImpl VolatileDBEnv{..} = do
+    mSt <- readTMVar _dbInternalState
+    case mSt of
+        Nothing -> EH.throwError' _dbErrSTM $ UserError ClosedDBError
+        Just st -> return $ \blockId ->
+            maybe (error msg) ibPreBid (Map.lookup blockId (_currentRevMap st))
```

Comment:
Just a note that `getSuccessors` follows a different approach than this: It will return a empty Set, if the blockId is not found, instead of throwing error. @mrBliss do you think we should merge the two approaches or you think this is different since it lives in `STM`?

Reply:
I have added a commit that lets `getSuccessors` follow the same approach.